ValheimPlus/Inventory.cs

```diff
@@ -1,19 +1,132 @@
-﻿using HarmonyLib;
-using System;
-using ValheimPlus.Configurations;
-
-namespace ValheimPlus
-{
-    [HarmonyPatch(typeof(Inventory), "IsTeleportable")]
-    public static class noItemTeleportPrevention
-    {
-        private static void Postfix(ref Boolean __result)
-        {
-            if (Configuration.Current.Items.IsEnabled)
-            {
-                if (Configuration.Current.Items.noTeleportPrevention)
-                    __result = true;
-            }
-        }
-    }
-}
+﻿using HarmonyLib;
+using System;
+using UnityEngine;
+using UnityEngine.UI;
+using ValheimPlus.Configurations;
+
+namespace ValheimPlus
+{
+    [HarmonyPatch(typeof(Inventory), "IsTeleportable")]
+    public static class noItemTeleportPrevention
+    {
+        private static void Postfix(ref Boolean __result)
+        {
+            if (Configuration.Current.Items.IsEnabled)
+            {
+                if (Configuration.Current.Items.noTeleportPrevention)
+                    __result = true;
+            }
+        }
+    }
+
+    [HarmonyPatch(typeof(Inventory), "TopFirst", new Type[] { typeof(ItemDrop.ItemData) })]
+    public static class changeInventoryFillOrder
+    {
+        public static bool Prefix(ref Boolean __result)
+        {
+            if (Configuration.Current.Inventory.IsEnabled &&
+                Configuration.Current.Inventory.inventoryFillTopToBottom)
+            {
+                __result = true;
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+    }
+
+    [HarmonyPatch(typeof(Inventory), MethodType.Constructor)]
+    [HarmonyPatch(new Type[] { typeof(string), typeof(Sprite), typeof(int), typeof(int) })]
+    public static class changeInventorySlotCount
+    {
+        static void Prefix(ref string name, ref Sprite bkg, ref int w, ref int h)

```

Comment:
We can follow convention here for any `ref` the parameter name should be prefixed with `__`. Ex: `ref string name` => `ref string __name`

Reply:
> We can follow convention here for any `ref` the parameter name should be prefixed with `__`. Ex: `ref string name` => `ref string __name`

As far as i know this is not possible. When patching a class constructor the parameters must be exact.
Here is the error:
[Error  :  HarmonyX] Failed to patch System.Void Inventory::.ctor(System.String name, UnityEngine.Sprite bkg, System.Int32 w, System.Int32 h): System.Exception: Parameter __name does not contain a valid index

Reply:
Oh I might be thinking of something else, sorry about that